dashboard/app/coverage.go

```diff
@@ -56,22 +56,94 @@ type funcStyleBodyJS func(
 	scope *coveragedb.SelectScope, onlyUnique bool, sss, managers []string, dataFilters cover.Format,
 ) (template.CSS, template.HTML, template.HTML, error)
 
+type coverageHeatmapParams struct {
+	manager    string
+	subsystem  string
+	onlyUnique bool
+	periodType string
+	nPeriods   int
+	dateTo     civil.Date
+	cover.Format
+}
+
+func makeHeatmapParams(ctx context.Context, r *http.Request) (*coverageHeatmapParams, error) {
+	// Prefixing "0" we don't fail on empty string.
+	minCoverLinesDrop, err := strconv.Atoi("0" + r.FormValue(covPageParams[MinCoverLinesDrop]))
+	if err != nil {
+		return nil, fmt.Errorf(covPageParams[MinCoverLinesDrop] + " should be integer")
+	}
+	onlyUnique := r.FormValue(covPageParams[UniqueOnly]) == "1"
+
+	periodType := r.FormValue(covPageParams[PeriodType])
+	if periodType == "" {
+		periodType = coveragedb.DayPeriod
+	}
+	if periodType != coveragedb.DayPeriod &&
```

Comment:
Silly question: can you use something like `!slices.Contains(allowedPeriods, periodType)` here instead? Will it be more idiomatic?